### PR TITLE
Add support for "change" trigger on text-area and text-field

### PR DESCRIPTION
**Pull request by baptistegrimaud**
In addition to the `blur` and `focus` triggers, add support for `change` so we can for instance, update a part of the UI when the user types in a text field or text area.

Partial implementation of https://github.com/Instawork/hyperview/issues/298 but doesn't handle the dirty/clean states.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "hyperview",
-  "version": "0.45.0",
```
Comment (adamstep) [inline package.json:3]: You don't need to update this version number, we do this automatically when pushing a release to npm (by running `yarn release:minor`). But that should wait after this code is approved and merged.

Comment (baptistegrimaud) [reply package.json]: reverted

**Comment by adamstep** [comment]
I think the implementation generally looks good, but I'd like to see some [examples added](https://github.com/Instawork/hyperview/tree/master/examples/behaviors) to the repo so we can test a few interactions. 

In particular, I want to verify the interaction of the `change` trigger with other attributes on the `<text-field>`, including `auto-focus` (should not trigger change), `mask` (if the change is not allowed, should we trigger?), and `secure-text` (change still triggers).

**Comment by baptistegrimaud** [comment]
> In particular, I want to verify the interaction of the `change` trigger with other attributes on the `<text-field>`, including `auto-focus` (should not trigger change), `mask` (if the change is not allowed, should we trigger?), and `secure-text` (change still triggers).

Thanks @adamstep for bringing up those cases.
I added some examples to cover them (see below), let me know if additional ones are needed!

https://user-images.githubusercontent.com/91907891/143160525-dbb3cfcf-669a-480e-aacc-b6b9c5d41e9b.mp4

**Comment by adamstep** [comment]
> PS: when a `mask` is provided, the `change` is still triggered, even if a non allowed character is entered. Is that acceptable or should I fix it?

This is what I was wondering about. I think the existing behavior is ok, but we should mention this interaction between `mask` and `change` trigger in the website docs.

Otherwise, the changes look good!

**Comment by baptistegrimaud** [comment]
> This is what I was wondering about. I think the existing behavior is ok, but we should mention this interaction between `mask` and `change` trigger in the website docs.

@adamstep , I just added [this commit](https://github.com/Instawork/hyperview/pull/360/commits/3b8a14f686b6e3e98d39a838a260d2f7f21a2ae0), let me know if I should word it differently :)
Thanks

**Comment by adamstep** [comment]
@baptistegrimaud I think it's good to use the same language as in the `mask` docs, maybe something like this:

> Note: when using the change trigger on a `<text-field>` that include the `mask` attribute, the behavior will be triggered even if the mask prevents the pressed key from being set.

**Review by adamstep — APPROVED** [review]
Nice work!

**Pull request merged**